File: javascript/lesson-03-tool-calling/src/services/order.service.ts
import { db } from "../database/db.js"

export const getOrderStatus = async (orderId: string) => {
    const result = await db.query(`
        SELECT id, status, carrier, estimated_delivery, total_amount 
        FROM orders
        WHERE id = $1`,
        [orderId]
    );

    if (result.rows.length === 0) {
        return { error: `No order found with ID ${orderId}` }
    }

    return result.rows[0];
};

export const listCustomerOrders = async (customerId: string) => {
    const result = await db.query(`
        SELECT id, status, total_amount, created_at
        FROM orders
        WHERE customer_id = $1
        ORDER BY created_at DESC`,
        [customerId]
    );

    return { orders: result.rows };
};

File: java/lesson-03-tool-calling/src/main/java/com/ailearning/toolcalling/service/OrderService.java
package com.ailearning.toolcalling.service;

import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.stereotype.Service;

import java.util.Map;

@Service
public class OrderService {
    private final JdbcTemplate jdbcTemplate;

    public OrderService(JdbcTemplate jdbcTemplate) {
        this.jdbcTemplate = jdbcTemplate;
    }

    public Map<String, Object> getOrderStatus(String orderId) {
        String sql = """
                SELECT id, status, carrier, estimated_delivery, total_amount
                FROM orders
                WHERE id = ?
                """;

        return jdbcTemplate.queryForMap(sql, orderId);
    }
}
